File: jasmine-standalone-2.4.1/spec/AirportSpec.js
describe ("airport when weather is not stormy", function() {

  var airport;
  var plane;

  beforeEach(function(){
    airport = new Airport();
    plane = new Plane();
    weather = new Weather();
    spyOn(Math, "random").and.returnValue(1);

  });

  it("default_capacity defaults to 5", function(){
    expect(airport.DEFAULT_CAPACITY).toEqual(5);
  })

  it("hanger is an emtpy array", function(){
    expect(airport.hanger.length).toEqual(0);
  })

  it("lands plane and increases docked planes by 1", function(){
    airport.land(plane);
    expect(airport.hanger).toContain(plane);
  })

  it("throws error if hanger is full", function(){
    for(i=1; i<=5;i++){airport.land(plane)};
    expect(function(){airport.land(plane)}).toThrow("Hanger is full");
  })

  it("cannot takeoff a plane when hanger is empty", function(){
    expect(function(){airport.takeoff(plane)}).toThrow("Hanger is empty");
  })

});

describe ("airport when weather is stormy", function() {

  var airport;
  var plane;

  beforeEach(function(){
    airport = new Airport();
    plane = new Plane();
    weather = new Weather();
    airport.land(plane);
    spyOn(Math, "random").and.returnValue(0);

  });

  it("cannot land plane", function(){
    expect(function(){airport.land(plane)}).toThrow("Weather is stormy");
  })

  it("cannot take off plane", function(){
    expect(function(){airport.takeoff(plane)}).toThrow("Weather is stormy");
  })

});

File: airport_spec.rb
require 'airport'
require 'plane'

describe Airport do
    describe '#land' do
        it "lands a plane at an airport" do
            airport = Airport.new(30)
            plane = Plane.new
            allow(airport).to receive(:stormy?).and_return false
            expect(airport) .to respond_to :land
        end 

        it "does not let plane land when capacity is full" do
            airport = Airport.new(30)
            plane = Plane.new
            allow(airport).to receive(:stormy?).and_return false
            30.times do 
            airport.land(plane)
            end 
            expect{airport.land(plane)}.to raise_error 'Capacity is full : Cannot land plane'
        end
    end
 
    describe'#take_off' do
        it "instructs a plane to take off" do
            airport = Airport.new(30)
            plane = Plane.new
            expect(airport).to respond_to :take_off
        end
    end

    describe '#weather' do
        it "does not allow planes to land when stormy" do
            airport = Airport.new(30)
            plane = Plane.new
            allow(airport).to receive(:stormy?).and_return true
            expect{airport.land(plane)}.to raise_error 'Weather is stormy : Cannot land plane'
        end

        it "does not allow planes to take off when stormy" do 
            airport = Airport.new(30)
            plane = Plane.new
            allow(airport).to receive(:stormy?).and_return true
            expect{airport.take_off(plane)}.to raise_error 'Weather is stormy : Plane cannot take off'
        end
    end


end 